wgpu-hal/src/gles/egl.rs

```diff
@@ -260,24 +264,48 @@ impl Inner {
         let (config, supports_native_window) = choose_config(&egl, display)?;
         egl.bind_api(egl::OPENGL_ES_API).unwrap();
 
+        let needs_robustness = true;
+        let mut khr_context_flags = 0;
+        let supports_khr_context = display_extensions.contains("EGL_KHR_create_context");
+
         //TODO: make it so `Device` == EGL Context
         let mut context_attributes = vec![
             egl::CONTEXT_CLIENT_VERSION,
             3, // Request GLES 3.0 or higher
         ];
-        // Debug requires EGL 1.5+
-        if flags.contains(crate::InstanceFlags::DEBUG) && version >= (1, 5) {
-            log::info!("\tEGL context: +debug");
-            //TODO: figure out why this is needed
-            context_attributes.push(egl::CONTEXT_OPENGL_DEBUG);
-            context_attributes.push(egl::TRUE as _);
+        if flags.contains(crate::InstanceFlags::DEBUG) {
+            if version >= (1, 5) {
+                log::info!("\tEGL context: +debug");
+                context_attributes.push(egl::CONTEXT_OPENGL_DEBUG);
+                context_attributes.push(egl::TRUE as _);
+            } else if supports_khr_context {
+                log::info!("\tEGL context: +debug KHR");
+                khr_context_flags |= EGL_CONTEXT_OPENGL_DEBUG_BIT_KHR;
+            } else {
+                log::info!("\tEGL context: -debug");
+            }
         }
-        if display_extensions.contains("EGL_EXT_create_context_robustness") {
-            log::info!("\tEGL context: +robust access");
-            context_attributes.push(egl::CONTEXT_OPENGL_ROBUST_ACCESS);
-            context_attributes.push(egl::TRUE as _);
+        if needs_robustness {
+            if version >= (1, 5) {
+                log::info!("\tEGL context: +robust access");
+                context_attributes.push(egl::CONTEXT_OPENGL_ROBUST_ACCESS);
+                context_attributes.push(egl::TRUE as _);
+            } else if supports_khr_context {
+                log::info!("\tEGL context: +robust access KHR");
+                khr_context_flags |= EGL_CONTEXT_OPENGL_ROBUST_ACCESS_BIT_KHR;
+            } else if display_extensions.contains("EGL_EXT_create_context_robustness") {
+                log::info!("\tEGL context: +robust access EXT");
+                context_attributes.push(EGL_CONTEXT_OPENGL_ROBUST_ACCESS_EXT);
```

Comment:
@kvark Thank you! But this not  fix #1577 on my Android device (8.0, EGL 1.4). Still got same `BadAttibute` error.

Reply:
@Gordon-F could you attach a new log to the issue with this PR? Curious about what path is taken.